src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -67,11 +67,13 @@ grpc_status_code TlsFetchKeyMaterials(
     const grpc_core::RefCountedPtr<grpc_tls_key_materials_config>&
         key_materials_config,
     const grpc_tls_credentials_options& options,
+    bool server_config,
```

Comment:
Please revert to the original API signature and instead, check `grpc_tls_credentials_options` to see if `GRPC_TLS_SKIP_ALL_SERVER_VERIFICATION` option is set and disable the check only if the option is set.